lib/ohai/dsl/plugin.rb

```diff
@@ -183,6 +185,18 @@ def method_missing(name, *args)
 
       private
 
+      def safe_get_attribute(*keys)
+        keys.inject(@data) do |attrs, key|
+          unless attrs.nil? || attrs.is_a?(Hash)
```

Comment:
Yeah but Mashes turn into Arrays once you traverse them, and `attrs[key]` is fine to run on an Array.

I think if you set it up and try to do something like

`safe_get_attributes("etc", "passwd", "group", "wheel", "members", 0)` 

It'll throw a TypeError instead of giving you back `"root"` (on my Mac anyway).